src/main/java/org/instituteatri/deep/service/ChatService.java

```diff
@@ -1,52 +1,126 @@
         messages.add(msg);
         LOGGER.info("Inserting {} to the database", msg);
-        messageService.save(msg, occurrenceId);
+        messageService.saveOllama(msg, occurrenceId);
         OllamaApi.ChatRequest build =
-                OllamaApi.ChatRequest.builder(model).withMessages(messages).build();
+                OllamaApi.ChatRequest.builder("llama3").withMessages(messages).build();
         ChatResponse response = new OllamaApi().chat(build);
         messages.add(Message.builder(Message.Role.ASSISTANT)
                 .withContent(response.message().content()).build());
         LOGGER.info("Inserting {} to the database", msg);
-        messageService.save(response.message(), occurrenceId);
+        messageService.saveOllama(response.message(), occurrenceId);
         return response;
     }
+
+    public ChatResponseDTO generateGemini(ChatRequestDTO request, String occurrenceId) {
+        OccurrenceResponseDTO occurrenceResponse = occurrenceService.getById(occurrenceId);
+        Occurrence occurrence = modelMapper.map(occurrenceResponse, Occurrence.class);
+        String url = "https://generativelanguage.googleapis.com/v1beta/models/gemini-pro:generateContent?key=" + GOOGLE_API_KEY;
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        List<Map<String, Object>> contents = new ArrayList<>();
+        occurrence.getMessages().forEach(x -> {
+            contents.add(Map.of("role", x.getRole().toString(), "parts", List.of(Map.of("text", x.getContent()))));
+        });
+        contents.add(Map.of("role", "user", "parts", List.of(Map.of("text", request.getMessage()))));
+
+        org.instituteatri.deep.model.Message msgUser = org.instituteatri.deep.model.Message.builder()
+                .role(Role.USER)
+                .occurrence(occurrence)
+                .content(request.getMessage())
+                .build();
+        messageService.saveGemini(msgUser);
+
+        Map<String, Object> requestBody = new HashMap<>();
+        requestBody.put("contents", contents);
+
+        HttpEntity<Map<String, Object>> requestEntity = new HttpEntity<>(requestBody, headers);
+        ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.POST, requestEntity, String.class);
+
+        String text = "";
+        if (response.getStatusCode() == HttpStatus.OK) {
+            String responseBody = response.getBody();
+            // Parse to json and get text
+            text = getString(responseBody, text);
+            org.instituteatri.deep.model.Message msgAI = org.instituteatri.deep.model.Message.builder()
+                    .role(Role.MODEL)
+                    .occurrence(occurrence)
+                    .content(text)
+                    .build();
+            messageService.saveGemini(msgAI);
+        } else {
+            System.out.println("Request failed with status code: " + response.getStatusCode());
```

Comment:
Replace System.out.println with LOGGER.error or a custom exception to maintain consistency and improve error control.